website/static/files/build_orca_reconcile.sql

```diff
@@ -60,9 +60,8 @@ BEGIN
   , last_update          	timestamp with time zone NOT NULL
   , size_in_bytes        	int8 NOT NULL
   , storage_class        	text NOT NULL
-  , delete_marker        	bool NOT NULL
-  , CONSTRAINT FK_reconcile_job_s3_objects FOREIGN KEY(job_id) REFERENCES reconcile_job(id)
-  , CONSTRAINT PK_reconcile_s3_object PRIMARY KEY(orca_archive_location, key_path)
```

Comment:
These keys cannot be added to partitions on the postgres version I have in AWS, though newer versions do support it. Thoughts?

Reply:
So I tested against the older version of PostgreSQL 10.14 In this case we need to add the keys to each partition. Below is a shortened test script I used to perform some validation.

```
-- Create job table
CREATE TABLE IF NOT EXISTS reconcile_job (
  id bigint NOT NULL PRIMARY KEY
);

INSERT INTO reconcile_job values (1);

-- Create reconcile_s3_object table
CREATE TABLE IF NOT EXISTS reconcile_s3_object
(
  job_id               	int8 NOT NULL
, orca_archive_location	text NOT NULL
, key_path             	text NOT NULL
            --, CONSTRAINT FK_reconcile_job_s3_objects FOREIGN KEY(job_id) REFERENCES reconcile_job(id)
            --, CONSTRAINT PK_reconcile_s3_object PRIMARY KEY(orca_archive_location, key_path)
)
PARTITION BY LIST (orca_archive_location);

-- Comment
COMMENT ON TABLE reconcile_s3_object IS 'Temporary table that holds the listing from the ORCA S3 bucket to use for comparisons against the ORCA catalog.';
COMMENT ON COLUMN reconcile_s3_object.job_id IS 'Job the S3 listing is a part of for the comparison. Foreign key to the reconcile jobs table.';
COMMENT ON COLUMN reconcile_s3_object.orca_archive_location IS 'ORCA S3 Glacier bucket name where the file is stored.';
COMMENT ON COLUMN reconcile_s3_object.key_path IS 'Full path and file name of the object in the S3 bucket.';


CREATE TABLE IF NOT EXISTS reconcile_s3_object_1
PARTITION OF reconcile_s3_object
    (
      CONSTRAINT PK_reconcile_s3_object_1 PRIMARY KEY(key_path)
    , CONSTRAINT FK_reconcile_job_s3_objects_1 FOREIGN KEY(job_id) REFERENCES reconcile_job(id)
    )
FOR VALUES IN ('a');
-- Comment
COMMENT ON TABLE reconcile_s3_object_1 IS 'Partition table for reconcile_s3_object based on orca_archive_location.';

CREATE TABLE IF NOT EXISTS reconcile_s3_object_2
PARTITION OF reconcile_s3_object
    (
      CONSTRAINT PK_reconcile_s3_object_2 PRIMARY KEY(key_path)
    , CONSTRAINT FK_reconcile_job_s3_objects_2 FOREIGN KEY(job_id) REFERENCES reconcile_job(id)
    )
FOR VALUES IN ('b');
-- Comment
COMMENT ON TABLE reconcile_s3_object_2 IS 'Partition table for reconcile_s3_object based on orca_archive_location.';
COMMIT;
```